client/src/components/StreetView.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { withGoogleMap, StreetViewPanorama } from 'react-google-maps';
+import Browser from '../util/browser';
+
+export type StreetViewAddr = {
+  bbl: string,
+  lat: number,
+  lng: number
+};
+
+export type StreetViewProps = {
+  addr: StreetViewAddr|null|undefined,
+};
+
+type ImplProps = {
+  heading: number,
+  coordinates: google.maps.LatLng|undefined
+};
+
+type State = ImplProps;
+
+const UnwrappedStreetViewImpl: React.FC<ImplProps> = props => {
+  return (
+    <StreetViewPanorama
+      visible
+      defaultPosition={props.coordinates}
+      position={props.coordinates}
+      pov={{ heading: props.heading, pitch: 15 }}
+      zoom={0.5}
+      options={{
+        disableDefaultUI: true,
+        enableCloseButton: false,
+        panControl: true,
+        fullscreenControl: true
+      }}
+    />
+  );
+};
+
+const StreetViewImpl = withGoogleMap(UnwrappedStreetViewImpl);
+
+export class StreetView extends React.Component<StreetViewProps, State> {
+  _isMounted: boolean = false;
+
+  constructor(props: StreetViewProps) {
+    super(props);
+    this.state = {
+      heading: 0,
+      coordinates: undefined
+    };
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+    this.updateHeadingAndCoordinates();
```

Comment:
The original `<DetailView>` that I've factored this component out of didn't actually call this code on mount... I suspect that `<DetailView>` may go through some initial prop churn that only made its `componentDidUpdate` matter, but because this new `<StreetView>` component has so few props, its `componentDidUpdate()` won't be called in the exact same way as `<DetailView>`'s did... whatever the actual cause, the heading/coordinates weren't being properly set in the initial render of this component, so I factored out the heading/coordinates logic into a separate function and called it in both `componentDidMount()` _and_ `componentDidUpdate()`.